### PR TITLE
Partial Revert "ENH: infer freq in timedelta_range (#32377)"

### LINKED ISSUE #35897: Regression in pandas 1.1.0: no longer allows adding a timestamp to timedelta range to create a datetime range.
Under pandas 1.0.5:

```python
>>> import pandas as pd
>>> pd.Timestamp.now() + pd.timedelta_range('0s','1s', periods=31)
DatetimeIndex([   '2020-08-25 12:27:57.690332',
               '2020-08-25 12:27:57.723665333',
               '2020-08-25 12:27:57.756998666',
                  '2020-08-25 12:27:57.790332',
               '2020-08-25 12:27:57.823665333',
               '2020-08-25 12:27:57.856998666',
                  '2020-08-25 12:27:57.890332',
               '2020-08-25 12:27:57.923665333',
               '2020-08-25 12:27:57.956998666',
                  '2020-08-25 12:27:57.990332',
               '2020-08-25 12:27:58.023665333',
               '2020-08-25 12:27:58.056998666',
                  '2020-08-25 12:27:58.090332',
               '2020-08-25 12:27:58.123665333',
               '2020-08-25 12:27:58.156998666',
                  '2020-08-25 12:27:58.190332',
               '2020-08-25 12:27:58.223665333',
               '2020-08-25 12:27:58.256998666',
                  '2020-08-25 12:27:58.290332',
               '2020-08-25 12:27:58.323665333',
               '2020-08-25 12:27:58.356998666',
                  '2020-08-25 12:27:58.390332',
               '2020-08-25 12:27:58.423665333',
               '2020-08-25 12:27:58.456998666',
                  '2020-08-25 12:27:58.490332',
               '2020-08-25 12:27:58.523665333',
               '2020-08-25 12:27:58.556998666',
                  '2020-08-25 12:27:58.590332',
               '2020-08-25 12:27:58.623665333',
               '2020-08-25 12:27:58.656998666',
                  '2020-08-25 12:27:58.690332'],
              dtype='datetime64[ns]', freq=None)
```

Under pandas 1.1.0:

```python
>>> import pandas as pd
>>> pd.Timestamp.now() + pd.timedelta_range('0s','1s', periods=31)
Traceback (most recent call last):
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/arrays/datetimelike.py", line 1157, in _validate_frequency
    raise ValueError
ValueError
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/indexes/extension.py", line 147, in method
    result = meth(_maybe_unwrap_index(other))
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/arrays/datetimelike.py", line 1446, in __radd__
    return self.__add__(other)
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/ops/common.py", line 65, in new_method
    return method(self, other)
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/arrays/datetimelike.py", line 1406, in __add__
    result = self._add_datetimelike_scalar(other)
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/arrays/timedeltas.py", line 441, in _add_datetimelike_scalar
    return DatetimeArray(result, dtype=dtype, freq=self.freq)
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/arrays/datetimes.py", line 284, in __init__
    type(self)._validate_frequency(self, freq)
  File "/Users/cbarber/miniconda3/envs/episodic-dev3/lib/python3.6/site-packages/pandas/core/arrays/datetimelike.py", line 1171, in _validate_frequency
    ) from e
ValueError: Inferred frequency None from passed values does not conform to passed frequency 33333333N
```

This error really makes no sense given the above code.

Obviously an unintended side-effect of some other fix.

### PR DESCRIPTION
**Pull request by simonjayhawkins**
closes #35897

maybe alternative to #36582 if no changes since #32377 now fail (not tested locally)

cc @phofl @jbrockmendel

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
> maybe alternative to #36582 if no changes since #32377 now fail (not tested locally)

looks like maybe just a doctest added since.

the release note will need to explain that the enhancement has been reverted, so a proper fix maybe more preferable.

will leave this a draft for now.

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
@simonjayhawkins yep looks good here too; if you can add a release note and merge master; ping on green (and you are in draft status)

**Comment by simonjayhawkins** [comment]
ci restarted

```
=========================== short test summary info ===========================
FAILED pandas/tests/plotting/test_groupby.py::TestDataFrameGroupByPlots::test_series_groupby_plotting_nominally_works
= 1 failed, 75523 passed, 1809 skipped, 1165 xfailed, 67 warnings in 1223.71s (0:20:23) =
```

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins yep looks good here too; if you can add a release note and merge master; ping on green (and you are in draft status)

@jreback green

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -620,10 +616,6 @@ def __floordiv__(self, other):
             if self.freq is not None:
                 # Note: freq gets division, not floor-division
                 freq = self.freq / other
-                if freq.nanos == 0 and self.freq.nanos != 0:
-                    # e.g. if self.freq is Nano(1) then dividing by 2
-                    #  rounds down to zero
-                    freq = None
```
Comment (jbrockmendel) [inline pandas/core/arrays/timedeltas.py:626]: is reverting this necessary?  im surprised there isnt a test that hits it

Comment (simonjayhawkins) [reply pandas/core/arrays/timedeltas.py]: no idea. maybe added to stop failing tests with inferred freq?

**Comment by jreback** [comment]
3rd time the charm, thanks @simonjayhawkins

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**